pkg/controller/kogitosupportingservice/kogitosupportingservice_controller.go

```diff
@@ -0,0 +1,239 @@
+		},
+		// Filter out ignoreServices
+		CreateFunc: func(e event.CreateEvent) bool {
+			supportingService := e.Object.(*v1alpha1.KogitoSupportingService)
+			return !contains(ignoreServices, supportingService.Spec.ServiceType)
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			supportingService := e.ObjectNew.(*v1alpha1.KogitoSupportingService)
+			return !contains(ignoreServices, supportingService.Spec.ServiceType)
+		},
+	}
+	err = c.Watch(&source.Kind{Type: &v1alpha1.KogitoSupportingService{}}, &handler.EnqueueRequestForObject{}, pred)
+	if err != nil {
+		return err
+	}
+
+	controllerWatcher := framework.NewControllerWatcher(r.(*ReconcileKogitoSupportingService).client, mgr, c, &v1alpha1.KogitoSupportingService{})
+	watchedObjects := []framework.WatchedObjects{
+		{
+			GroupVersion: routev1.GroupVersion,
+			AddToScheme:  routev1.Install,
+			Objects:      []runtime.Object{&routev1.Route{}},
+		},
+		{
+			GroupVersion: imgv1.GroupVersion,
+			AddToScheme:  imgv1.Install,
+			Objects:      []runtime.Object{&imgv1.ImageStream{}},
+		},
+		{
+			Objects:      []runtime.Object{&v1alpha1.KogitoInfra{}},
+			EventHandler: &handler.EnqueueRequestForOwner{IsController: false, OwnerType: &v1alpha1.KogitoSupportingService{}},
+			Predicate: predicate.Funcs{
+				UpdateFunc: func(e event.UpdateEvent) bool {
+					return isKogitoInfraUpdated(e.ObjectOld.(*v1alpha1.KogitoInfra), e.ObjectNew.(*v1alpha1.KogitoInfra))
+				},
+			},
+		},
+		{Objects: []runtime.Object{&corev1.Service{}, &appsv1.Deployment{}, &corev1.ConfigMap{}}},
+	}
+	if err = controllerWatcher.Watch(watchedObjects...); err != nil {
+		return err
+	}
+	return nil
+}
+
+// blank assignment to verify that ReconcileKogitoSupportingService implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileKogitoSupportingService{}
+
+// ReconcileKogitoSupportingService reconciles a KogitoSupportingService object
+type ReconcileKogitoSupportingService struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client *client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a KogitoSupportingService object and makes changes based on the state read
+// and what is in the KogitoSupportingService.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileKogitoSupportingService) Reconcile(request reconcile.Request) (result reconcile.Result, resultErr error) {
+	// Fetch the KogitoSupportingService instance
+	instance, resultErr := fetchKogitoSupportingService(r.client, request.Name, request.Namespace)
+	if resultErr != nil {
+		return
+	}
+
+	// Do not trigger the Reconcile of kogitoSupportingController for ignored services
+	if contains(ignoreServices, instance.Spec.ServiceType) {
```

Comment:
So you're ignoring Data Index, right? Not sure if this is the best approach, but I understand you guys did a hell of a job testing all sorts of integration, 'cause we have that CM reconciliation problem, right? I believe we can have this for now with the timebox we have, but would be nice to have a follow up JIRA to have everything been handled by one controller instead. We might need to revisit the required use cases for Data Index.

Reply:
Yes, it is to ignore DataIndex here, created [KOGITO-3682](https://issues.redhat.com/browse/KOGITO-3682) for further evaluation.